packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -147,19 +188,92 @@ export const createSchema = (schema: Schema, config?: DataSourceGenerateConfig):
     throw new Error('No valid tables found. Make sure at least one table has a primary key.');
   }
 
+  const engine = schema.getEngine().type;
+
+  const nullEnums = schema  // find null enums in all models
+    .getModels()
+    .map((model) => model
+        .getFields()
+        .filter((field) => field.type.kind === 'Enum')
+        .map((field) => {
+          if(field.type.kind === 'Enum'){
+            return createEnums(field.type, engine, model.getName(), field.name);
+          }
+          else{
+            return undefined;
+          }
+     }));
+
+    const notNullEnums = schema  // find Non null enums in all models
+    .getModels()
+    .map((model) => model
+        .getFields()
+        .filter((field) => field.type.kind === 'NonNull' && field.type.type.kind === 'Enum')
+        .map((field) => {
+          if(field.type.kind === 'NonNull' && field.type.type.kind === 'Enum'){
+            return createEnums(field.type.type, engine, model.getName(), field.name);
+          }
+          else{
+            return undefined;
+          }
+     }));
+
   const models = schema
     .getModels()
     .filter((model) => model.getPrimaryKey())
     .map((model) => {
-      return createModel(model);
+      return createModel(model, engine);
     });
+
+  const combinedEnums = nullEnums.concat(notNullEnums).flat() // making 1 D array
+
+  if (engine === 'Postgres') { // for postgresql
```

Comment:
For future: Instead of introducing these engine based conditionals that bloats up this method, we could move these to separate methods inside an engine specific controller.

Reply:
Sure I will look into it.

Reply:
@phani-srikar I have updated the implementation to make it less bloated in the current task as well and more strongly typed.